File: Python/project/src/Lists/FindMissingElements/FindMissingElements.py
from typing import List


def main():
    input = [[1,4,2,5], [7,8,6,9], [5,1]]
    for i, row in enumerate(input):
        solution = Solution()
        print("findMissingElements: " + str(solution.findMissingElements(row)) + "\n")


class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        output = []
        nums.sort()
        print("nums: " + str(nums))

        min = nums[0]
        max = nums[len(nums)-1]

        for num in range(min+1, max):
            print("     i: " + str(num))
            if num not in nums:
                output.append(num)


        return output


if __name__=="__main__":
    main()

File: Java/src/main/java/Lists/FindMissingElements/FindMissingElements.java
package Lists.FindMissingElements;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FindMissingElements {

    public static void main(String [] args) {
        int [][] input = {{1,4,2,5}, {7,8,6,9}, {5,1}};
        for (int [] row : input) {
            System.out.println("Find Missing Elements: " + findMissingElements(row) + "\n");
        }
    }

    public static List<Integer> findMissingElements(int[] nums) {
        List<Integer> output = new ArrayList<>();

        Arrays.sort(nums);
        int min = nums[0];
        int max = nums[nums.length-1];

        int current = min + 1;
        while (current < max) {
            if (Arrays.binarySearch(nums, current) < 0) {
                output.add(current);
            }

            ++current;
        }


        return output;
    }


}
